tracks/rust/exercises/pythagorean-triplet/mentoring.md

```diff
@@ -0,0 +1,74 @@
+### Concepts
+
+- Option
+- Math
+
+### Reasonable solutions
+
+A reasonable solution should do the following:
+
+- Use `a.pow(2)` rather than `a*a`
+- not use return for the last line of the function
+- Not use floating point values, which introduce the posibility of rounding errors
+- Not introduce posibility of integer underflow.
+- Accounts for possibly not finding a triplet and returns None, not Some(0)
+
+A good solution may go above and beyond and also do the following:
+
+- Solve the problem with euler's formula
+- Break out the core logic of find into a variant that takes the perimeter as an argument
+- Introduce negative test cases.
+
+### Examples
+
+restricting iteration to only be over valid triangles
+
+```rust
+use std::cmp::max;
+use std::collections::HashSet;
+
+pub fn find(sum: u32) -> HashSet<[u32; 3]> {
+    let mut results = HashSet::new();
+
+    for a in 2..(sum / 3) {
+        let bmin = sum / 2 - a + 1;
+        let cmax = sum - bmin - a;
+        let diff = cmax - bmin;
+
+        for b in max(bmin, a)..=bmin + (diff / 2) {
+            let c = sum - b - a;
+            if a * a + b * b == c * c {
+                results.insert([a, b, c]);
+            }
+        }
+    }
+
+    results
+}
+```
+
+### Example Comments
+
+If they use return on the last line for the None variant.
```

Comment:
There is no room for a `None` variant in your example and in the function stub delivered to the student.